src/Transport/Addressing/Sanitization/Strategies/EndpointOrientedTopologySanitization.cs

```diff
@@ -4,21 +4,13 @@
 
     public class EndpointOrientedTopologySanitization : ISanitizationStrategy
     {
-        IValidationStrategy validationStrategy;
-
-        public EndpointOrientedTopologySanitization(IValidationStrategy validationStrategy)
-        {
-            this.validationStrategy = validationStrategy;
-        }
-
-        public string Sanitize(string entityPathOrName, EntityType entityType)
+        public string Sanitize(string entityPathOrName, EntityType entityType, ValidationResult validationResult)
         {
             // remove invalid characters
             var rgx = new Regex(@"[^a-zA-Z0-9\-\._]");
```

Comment:
`EntityNameValidationV6Rules` and `EndpointOrientedTopologySanitization` share the same `regex` to check and sanitize `pathOrName`. This is another symptom that validation and sanitization are more tiled than we think.
Does make sense validate with a regex and sanitize with another? IMO no